addons/sale/models/sale_order.py

```diff
@@ -890,12 +890,15 @@ def action_confirm(self):
 
         self.with_context(context)._action_confirm()
 
-        if self[:1].create_uid.has_group('sale.group_auto_done_setting'):
-            # Public user can confirm SO, so we check the group on any record creator.
-            self.action_done()
+        self.filtered(lambda so: so._should_be_locked()).action_done()
 
         return True
 
+    def _should_be_locked(self):
```

Comment:
the name is a little confusing (since in the code it's named "done" not locked), but in 17.0 it is named locked so I guess it's simpler to have the same method name in both versions